### PR TITLE
arrays-arrays_with_methods-hard-q10

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Update test cases to use arrayUtil

**Comment by SamamthaH** [comment]
> Update test cases to use arrayUtil

Is that really necessary? The point of arrayUtil is that it can make very large test cases, but I already have a test case in this that is 1000 size

File: src/arrays/arrays_with_methods/hard/q10/ArrayofDigits.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package arrays.arrays_with_methods.hard.q10;
+
+/*
+Write a Java program that asks the user to enter two integers, where the first number n represents the numeric value to
+be stored and the second number k represents the repetitions of the first number. Then, call a method that you will also
+need to define called digitInt, which returns an integer array of size k all initialized to the value of n. Lastly,
+in the main method, display the resulting array that was outputted from the method call.
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q10/ArrayofDigits.java:7]:
```suggestion
in the main method, print the array which is returned from the digitInt method.
```

File: src/arrays/arrays_with_methods/hard/q10/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                {b, int.class}
+        };
+        MethodTest m = new MethodTest(ArrayofDigits.class, "digitInt", arguments);
+        Object output = m.callMethod();
+
+        int[] arr = new int[b];
+        for (int x = 0; x < b; x++)
+            arr[x] = a;
+
+        CustomAssertions._assertArrayEquals(arr, output, "Your digitInt method does not return an array of the specified integers to the specified length.");
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q10/MainTest.java:72]:
```suggestion
        CustomAssertions._assertArrayEquals(arr, output, "Your digitInt method does not return the correct array based on the input parameters.");
```

File: src/arrays/arrays_with_methods/hard/q10/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void correctMainMethodOutput(String in, int a, int b) throws InvalidClauseException {
+        Clause[][] c = new Clause[1][b * 2];
+        int count = 0;
+        for (int x = 0; x < b; x++) {
+            c[0][count++] = new IntegerLiteral(a);
+            c[0][count++] = new NewLine();
+        }
+
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q10/MainTest.java:84]: We should add a custom error message here I think

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q10/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        for (int x = 0; x < b; x++)
+            arr[x] = a;
+
+        CustomAssertions._assertArrayEquals(arr, output, "Your digitInt method does not return the correct based on the input parameters.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void correctMainMethodOutput(String in, int a, int b) throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program does not correctly print the array of integers.";
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q10/MainTest.java:78]:
```suggestion
        TestOption.incorrectStructureErrorMessage = "Your program does not correctly print the generated array of integers.";
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Pull request merged**